src/app/level2/level2.component.ts

```diff
@@ -6,12 +6,75 @@ import { Component } from '@angular/core';
 })
 export class Level2Component {
   public currentPlayerIndex = 1;
+  private currentWinnerIndex = 0;
+  private playerStyles: string[];
+  private playerNames: string[];
+  protected gameBoard!: number[][];
+
+  constructor() {
+    this.playerNames = ['', 'Red', 'Blue']
+    this.playerStyles = ['', 'occupied-1', 'occupied-2'];
+    this.restart();
+  }
+
+  public get winner() {
+    return this.playerNames[this.currentWinnerIndex]
+  }
 
   public drop(colIx: number) {
+    if (!!this.currentWinnerIndex) {
```

Comment:
This works, but I would prefer `this.currentWinnerIndex !== 0`. I find it less "overly smart" and easier to understand for other people. However, your code works, too.